app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -163,12 +202,44 @@ def parse_deployment_name(name)
         name.sub(/^.*deployment=/, '')
       end
 
-      def parse_middleware_server(eap)
+      def parse_server_group_name(name)
+        name.sub(/^Domain Server Group \[/, '').chomp(']')
+      end
+
+      def parse_domain_server_name(name)
+        name.sub(/^.*\/server=/, '')
+      end
+
+      def parse_middleware_domain(domain)
+        {
+          :feed       => domain.feed,
+          :ems_ref    => domain.path,
+          :nativeid   => domain.id,
+          :name       => domain.properties['Name'],
+          :type_path  => domain.type_path,
+          :properties => domain.properties
+        }
+      end
+
+      def parse_middleware_server_group(domain, group)
+        {
+          :feed              => group.feed,
+          :ems_ref           => group.path,
+          :nativeid          => group.id,
+          :middleware_domain => domain,
+          :name              => parse_server_group_name(group.name),
+          :type_path         => group.type_path,
+          :profile           => group.properties['Profile'],
+          :properties        => group.properties
+        }
+      end
+
+      def parse_middleware_server(eap, name = nil)
         {
           :feed       => eap.feed,
           :ems_ref    => eap.path,
           :nativeid   => eap.id,
-          :name       => parse_name(eap.id),
+          :name       => name || parse_name(eap.id),
```

Comment:
great, thanks. worth to add it as a comment in the code for those who won't see this correspondence when reading the merged code.